### ISSUE TITLE
Subfolder scanning

### ISSUE THREAD

**Issue by pahakalle** [gorgarp/BatchConverter#9]
Usually my files are all in one folder in subdirectories. For example main directory is "Movies" and bellow it is  "Doctor who" and in that directory is either the files or there could be "Season X" directories bellow that.

**Describe the solution you'd like**
Add a scan subdirectories functionality either on by default or as an argument.

**Additional context**
As the current code uses `find` to find the files this [site](https://unix.stackexchange.com/a/93326) could be helpful, if this type of feature would be considered to be added.

**Comment by gorgarp** [comment]
It should be recursive as is, since the -iname values are in quotes, which prevents glob expansion.

Are you telling me it isn't currently? If so, I might have to fix some escapes etc

**Comment by Wolveix** [comment]
Yeah, there are a few variables that haven't been quoted. Check your script against https://www.shellcheck.net

**Comment by gorgarp** [comment]
I see.

Want to take a look at testing branch?

**Comment by Wolveix** [comment]
Looks better. Though, you should change `"$(echo "$codec_name")"` to `"$codec_name"` and replace `echo "--------- Converting files ----------"` with `printf "\\n--------- Converting files ----------\\n"` (remove the redundant echo before and change the ending text to this format as well).

**Comment by gorgarp** [comment]
Good call. Thanks!

Not a coder by trade, actually a Sysadmin for virtualization. Just enough to be dangerous!

Take another look if/when you have time

**Comment by Wolveix** [comment]
No worries :) 

Ah, I'm both a developer and systems administrator. You should check out https://hostballs.com some time!

You're going to want to change this `if [[ ! "$("$codec_name")" == "codec_name=h264 codec_name=aac" ]];` to this `if [[ ! "$codec_name" == "codec_name=h264 codec_name=aac" ]];`. Also, remove the `echo ""` before the first `printf` call.

**Comment by gorgarp** [comment]
Haha yeah, still getting up to speed for sure. Came up through Windows team and now doing a bunch of Federal Regulatory stuff. Not much ability to do off the cuff scripting in that world! This really started because I have a private plex with about 27 TB of content, haphazardly laid out over the years. Thought I'd share the effort because I can't be the only one!

Done! Thanks.

I'm working with a buddy of mine to add some more features. Stay tuned!

**Comment by Wolveix** [comment]
Ah nice, my collection just surpassed 55TB 😛 

I'll be sure to keep an eye on the repo 😎 I probably won't use the project myself though since I've already got my own tool. All of your stated planned future features are already implemented there, so please do feel free to use some of my source code if you're ever stuck https://github.com/Wolveix/Plexus 😄 Currently, your script doesn't account for subtitles, plus you should really add `-map 0` to your FFmpeg call!

**Comment by gorgarp** [comment]
Actually, I just looked at yours and it does indeed have a lot of functions I've been thinking of....

Are you open to collaborating some? If so, I'll just kill this and reroute my postings to yours...

**Comment by Wolveix** [comment]
Oh yeah absolutely :) I did suggest this in the issue I opened, but I guess you weren't notified when you closed it haha.

**Comment by gorgarp** [comment]
Hahaha my bad! Yeah, give me a bit and I'll go through reddit / emby / plex to redirect.

I'll change my readme to refer over as well.

Be in touch soon!

**Comment by Wolveix** [comment]
Sounds great!
